File: UsagiKanji/Frontend/src/types/kanji.ts
export interface KanjiListItem {
    id: string;
    character: string;
    primaryMeaning?: string;
    isLearned: boolean;
}

export interface PaginatedList<T> {
    items: T[];
    pageIndex: number;
    totalPages: number;
}

export interface KanjiListParams {
    pageIndex: number;
    pageSize: number;
    sortBy?: string;
}

export type ReadingType = "Onyomi" | "Kunyomi";

export interface Reading {
    value: string;
    type: ReadingType;
}

export interface Meaning {
    value: string;
    isPrimary: boolean;
}

export interface VocabularyItem {
    text: string;
    common: boolean;
    kanaReadings: string[];
    glosses: string[];
}

export interface KanjiDetail {
    id: string;
    character: string;
    strokeCount: number;
    grade: number;
    sortIndex_Grade?: number;
    jlptLevel?: number;
    sortIndex_JLPT?: number;
    frequencyRank?: number;
    heisigNumber?: number;
    heisig6Number?: number;
    readings: Reading[];
    meanings: Meaning[];
    vocabulary: VocabularyItem[];
    notes?: string;
    keyword?: string;
    isLearned: boolean;
}

File: UsagiKanji/Backend/Domain/Entities/Kanji.cs
namespace Domain.Entities
{
    public class Kanji : BaseEntity
    {
        public string Character { get; set; } = string.Empty;
        public int? StrokeCount { get; set; }
        public int? Grade { get; set; }
        public int? SortIndex_Grade { get; set; }
        public int? SortIndex_JLPT { get; set; }
        public int? JLPTLevel { get; set; }
        public int? FrequencyRank { get; set; }
        public int? HeisigNumber { get; set; }
        public int? Heisig6Number { get; set; }
        
        

        public ICollection<Reading> Readings { get; set; } = new List<Reading>();
        public ICollection<Meaning> Meanings { get; set; } = new List<Meaning>();
        public ICollection<VocabularyKanjiCharacter> VocabularyKanjiCharacters { get; set; } = new List<VocabularyKanjiCharacter>();
        public ICollection<UserKanji> UserKanjis { get; set; } = new List<UserKanji>();

    }
}
